packages/core/src/Machine.ts

```diff
@@ -49,13 +50,16 @@ export function Machine<
 }
 
 export function createMachine<
-  TModel extends Model<any, any, any>,
+  TModel extends Model<any, any, any, any>,
   TContext = ModelContextFrom<TModel>,
   TEvent extends EventObject = EventFrom<TModel>,
-  TTypestate extends Typestate<TContext> = { value: any; context: TContext }
+  TTypestate extends Typestate<TContext> = { value: any; context: TContext },
+  TAction extends ActionObject<TContext, TEvent> = ModelActionsFrom<TModel>
```

Comment:
`ActionObject` has an `exec` property - it seems to me that this wouldn't be desired here, as this type is meant to only represent "action descriptions". What do you think about changing the constraint to the `BaseActionObject`? Where maybe the name of this type could be improved in the process (not sure if using `BaseActionObject` would fit this location here best)

Reply:
`BaseActionObject` makes sense as the base for other action objects; let's keep the name for now. It's not meant to be used externally anyway.

Reply:
What about changing the `extends` constraint here?

Reply:
To `BaseActionObject`? Works for me